com.microsoft.mrtk.uxcomponents/Editor/CreateElementMenus.cs

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using UnityEditor;
+using UnityEngine;
+using UnityEngine.UI;
+using UnityEditor.UI;
+using UnityEngine.EventSystems;
+using TMPro;
+using System.Reflection;
+using System.Linq;
+
+namespace Microsoft.MixedReality.Toolkit.Editor
+{
+    /// <summary>
+    /// Add-GameObject-menus for creating common MRTK UX components. Most of these
+    /// spawn prefabs, but some are just ad-hoc objects created in code (like panels,
+    /// backplates, etc).
+    /// </summary>
+    static internal class CreateElementMenus
+    {
+        // EmptyButton.prefab
+        // A completely empty button, with no icon, text, label, etc.
+        private static readonly string EmptyButtonPath = AssetDatabase.GUIDToAssetPath("b85e005d231192249b7077b40a4d4e45");
+
+        // ActionButton.prefab
+        // The basic building block button; contains an icon, text, and label.
+        private static readonly string ActionButtonPath = AssetDatabase.GUIDToAssetPath("c6b351a67ceb69140b199996bbbea156");
+
+        // Reflection into internal UGUI editor utilities.
+        private static System.Reflection.MethodInfo PlaceUIElementRoot = null;
+
+        private static GameObject CreateElement(string path, MenuCommand menuCommand)
+        {
+            Object prefab = AssetDatabase.LoadAssetAtPath(path, typeof(Object));
+            GameObject gameObject = PrefabUtility.InstantiatePrefab(prefab) as GameObject;
+            Undo.RegisterCreatedObjectUndo(gameObject, "Create " + gameObject.name);
+
+            // This is evil :)
+            // UGUI contains plenty of helper utilities for spawning and managing new Canvas objects
+            // at edit-time. Unfortunately, they're all internal, so we have to use reflection to
+            // access them.
+            if (PlaceUIElementRoot == null)
+            {
+                // We're using SelectableEditor type here to grab the assembly instead of going
+                // and hunting down the assembly ourselves. It's a bit more convenient and durable.
+                PlaceUIElementRoot = typeof(SelectableEditor).Assembly.GetType("UnityEditor.UI.MenuOptions").GetMethod(
```

Comment:
Might want another `null` check after this with an error log, largely to catch the case where Unity renames the namespace/type/method/whatever in a future Editor version, which they like to do 🙃

Reply:
Done